src/platform/plugins/shared/dashboard/public/dashboard_api/settings_manager.tsx

```diff
@@ -7,130 +7,115 @@
-      syncColors: syncColors$.value,
-      syncCursor: syncCursor$.value,
-      syncTooltips: syncTooltips$.value,
-      tags: tags$.value,
-      timeRestore: timeRestore$.value ?? DEFAULT_DASHBOARD_STATE.timeRestore,
-      useMargins: useMargins$.value,
+      ...(description && { description }),
+      tags,
+      timeRestore,
+      title,
+      options,
     };
   }
 
-  function setSettings(settings: DashboardSettings) {
-    setSyncColors(settings.syncColors);
-    setSyncCursor(settings.syncCursor);
-    setSyncTooltips(settings.syncTooltips);
-    setTags(settings.tags);
-    setTimeRestore(settings.timeRestore);
-    setUseMargins(settings.useMargins);
-    titleManager.api.setHideTitle(settings.hidePanelTitles);
-    titleManager.api.setDescription(settings.description);
-    titleManager.api.setTitle(settings.title);
-  }
-
-  const comparators: StateComparators<DashboardSettings> = {
-    title: titleComparators.title,
-    description: titleComparators.description,
-    hidePanelTitles: 'referenceEquality',
-    syncColors: 'referenceEquality',
-    syncCursor: 'referenceEquality',
-    syncTooltips: 'referenceEquality',
-    timeRestore: 'referenceEquality',
-    useMargins: 'referenceEquality',
-    tags: 'deepEquality',
-  };
-
   return {
     api: {
-      ...titleManager.api,
-      getSettings,
+      description$: stateManager.api.description$,
+      getSettings: stateManager.getLatestState,
+      hideTitle$: stateManager.api.hidePanelTitles$,
       settings: {
-        syncColors$,
-        syncCursor$,
-        syncTooltips$,
-        useMargins$,
+        syncColors$: stateManager.api.syncColors$,
+        syncCursor$: stateManager.api.syncCursor$,
+        syncTooltips$: stateManager.api.syncTooltips$,
+        useMargins$: stateManager.api.useMargins$,
       },
-      setSettings,
-      setTags,
-      timeRestore$,
+      setSettings: stateManager.reinitializeState,
+      setTags: stateManager.api.setTags,
+      timeRestore$: stateManager.api.timeRestore$,
+      title$: stateManager.api.title$,
     },
     internalApi: {
+      serializeSettings,
       startComparing$: (lastSavedState$: BehaviorSubject<DashboardState>) => {
-        return combineLatest([
-          syncColors$,
-          syncCursor$,
-          syncTooltips$,
-          tags$,
-          timeRestore$,
-          useMargins$,
-
-          titleManager.anyStateChange$,
-        ]).pipe(
+        return stateManager.anyStateChange$.pipe(
           debounceTime(100),
-          map(() => getSettings()),
+          map(() => stateManager.getLatestState()),
           combineLatestWith(lastSavedState$),
-          map(([latestState, lastSavedState]) =>
-            diffComparators(comparators, lastSavedState, latestState)
-          )
+          map(([latestState, lastSavedState]) => {
+            const { description, tags, timeRestore, title, ...optionDiffs } = diffComparators(
```

Comment:
nit: Why use `diffComparators` here rather than `areComparatorsEqual`? If we need to return all values anyway can't we return everything is any comparator fails? And we can even spread the defaults onto it.